dispatches/models/nuclear_case/chgtank.py

```diff
@@ -0,0 +1,590 @@
+for holdup in the custom material and energy balances""",
+        ),
+    )
+    config.declare("momentum_balance_type", ConfigValue(
+        default=MomentumBalanceType.pressureTotal,
+        domain=In(MomentumBalanceType),
+        description="Momentum balance construction flag",
+        doc="""Indicates what type of momentum balance should be constructed,
+**default** - MomentumBalanceType.pressureTotal.
+**Valid values:** {
+**MomentumBalanceType.none** - exclude momentum balances,
+**MomentumBalanceType.pressureTotal** - single pressure balance for material,
+**MomentumBalanceType.pressurePhase** - pressure balances for each phase,
+**MomentumBalanceType.momentumTotal** - single momentum balance for material,
+**MomentumBalanceType.momentumPhase** - momentum balances for each phase.}"""))
+    config.declare("has_pressure_change", ConfigValue(
+        default=True,
+        domain=In([True]),
+        description="Pressure change term construction flag",
+        doc="""Indicates whether terms for pressure change should be
+constructed,
+**default** - True. Must always be true for tank operations
+**Valid values:** {
+**True** - include pressure change terms}"""))
+    config.declare("property_package", ConfigValue(
+        default=useDefault,
+        domain=is_physical_parameter_block,
+        description="Property package to use for control volume",
+        doc="""Property parameter object used to define property calculations,
+**default** - useDefault.
+**Valid values:** {
+**useDefault** - use default package from parent model or flowsheet,
+**PhysicalParameterObject** - a PhysicalParameterBlock object.}"""))
+    config.declare("property_package_args", ConfigBlock(
+        implicit=True,
+        description="Arguments to use for constructing property packages",
+        doc="""A ConfigBlock with arguments to be passed to a property block(s)
+and used when constructing these,
+**default** - None.
+**Valid values:** {
+see property package for documentation.}"""))
+
+
+@declare_process_block_class("CHGTank",
+                             doc="Simple compressed hydrogen gas tank model")
+class CHGTankData(UnitModelBlockData):
+    """
+    Simple compressed hydrogen gas tank model.
+    Unit model to store or supply compressed hydrogen gas.
+    
+    """
+    CONFIG = ConfigBlock()
+    _make_chg_tank_config_block(CONFIG)
+
+    def build(self):
+        """Building model
+        Args:
+            None
+        Returns:
+            None
+        """
+        super(CHGTankData, self).build()
+
+        # Build Control Volume
+        self.control_volume = ControlVolume0DBlock(default={
+            "dynamic": self.config.dynamic,
+            "property_package": self.config.property_package,
+            "property_package_args": self.config.property_package_args})
+
+        # add inlet and outlet states
+        self.control_volume.add_state_blocks(has_phase_equilibrium=False)
```

Comment:
Maybe something to think about later, but should we allow for the possibility of phase change?